Runner/Program.cs

```diff
@@ -38,74 +38,72 @@ public static void Main(string[] args)
                 .AddJsonFile("appsettings.Development.json", optional: true)
                 .Build();
 
-            using (var di = BuildDependencyInjection(config))
+            using var di = BuildDependencyInjection(config);
+            var logger = di.GetService<ILogger<Program>>();
+            var githubConfig = di.GetService<GitHubConfiguration>();
+            var ghClient = di.GetService<IGitHubClient>();
+            var repositoryValidator = di.GetService<RepositoryValidator>();
+            var client = di.GetService<ValidationClient>();
+            client.Init().Wait();
+
+            void scanner(IEnumerable<string> repositories, Options options)
             {
-                var logger = di.GetService<ILogger<Program>>();
-                var githubConfig = di.GetService<GitHubConfiguration>();
-                var ghClient = di.GetService<IGitHubClient>();
-                var repositoryValidator = di.GetService<RepositoryValidator>();
-                var client = di.GetService<ValidationClient>();
-                client.Init().Wait();
-
-                void scanner(IEnumerable<string> repositories, Options options)
+                var start = DateTime.UtcNow;
+                var results = repositories.Select(repo =>
                 {
-                    var start = DateTime.UtcNow;
-                    var results = repositories.Select(repo =>
-                    {
-                        // This sleeps used to avoid hitting GitHub API limits
-                        Thread.Sleep(TimeSpan.FromSeconds(3));
-                        return client.ValidateRepository(githubConfig.Organization, repo, options.IgnoreRepositoryRules).ConfigureAwait(false).GetAwaiter().GetResult();
-                    }).ToArray();
+                    // This sleeps used to avoid hitting GitHub API limits
+                    Thread.Sleep(TimeSpan.FromSeconds(3));
```

Comment:
I think these can be refactore in more elegant form with async main `public static async Task Main(string[] args)`. Replace thread.sleep -> task.Delay in that case.

Reply:
Good idea.